geth/common/types.go

```diff
@@ -43,19 +43,19 @@ func (k *SelectedExtKey) Hex() string {
 	return k.Address.Hex()
 }
 
-// MessageStat defines a struct to hold given facts about a message stat.
-type MessageStat struct {
-	Type            string             `json:"type"`
-	Protocol        string             `json:"protocol"`
-	Status          string             `json:"status"`
-	Envelope        []byte             `json:"envelope"`
-	TimeSent        uint32             `json:"time,omitempty"`
-	Payload         []byte             `json:"payload,omitempty"`
-	Hash            string             `json:"envelope_hash"`
-	FromDevice      string             `json:"from_device,omitempty"`
-	ToDevice        string             `json:"to_device,omitempty"`
-	RejectionReason error              `json:"rejection_reason,omitempty"`
-	Source          whisper.NewMessage `json:"source,omitempty"`
+// MessageState defines a struct to hold given facts about a message stat.
+type MessageState struct {
+	Type            string             `json:"type"`                       // Defines Direction type: Incoming or Outgoing.
```

Comment:
Comments behind the fields make the line pretty wide, especially with annotations like here. So I would prefer it more `godoc`. So e.g. here

```
// Type defines the message direction as "Incoming" or "Outgoing".
Type string `json:"type"`
```

Btw, are the directions really written in camel-case?